### DISCUSSION TITLE
Worldedit Selection for all

### DISCUSSION THREAD

**Discussion by ghost** [Zrips/CMI#3961] [Feature Requests]
Can you please add Worldedit Selection for every Thing on CMI. Like Portals, Jails and so on...

The Integrated Selection is not good. With worldedit i can faster and easier select things.  And we can use different Visualizers.

**Comment by CloudeLecaw** [comment]
If you wish to visulice thebselection you can use the plugin SelectionVirtualizer. you can buy it on spigot if you want ;) 
About having one wand for WorldEdit and cmi... it would just confuse with world edit selection Tool if two tools try the same. Cmi allready has a selection wand for this all like portals etc. (wooden shovel be default) its a good think that two it has its own selection on tool.

**Comment by ghost** [comment]
Selection Visualizer is not good. It has many features but the particle display is not good. I know that you can customize it but you won't get the exact result from other Visualizers. And I use CUI as visualizer via Fabric Worldedit and there is no plugin that works anyway. 
I wonder why CMI has its own selection tool anyway. Worldedit has to be the standard because every smart server has it installed

**Comment by CloudeLecaw** [comment]
It has to be? X-D it up ti Zrips what has to be and what not. This way it works now does not need worldedit as dependence to have a selection tool

**Comment by ghost** [comment]
Name me a server that does not use Worldedit but something as extensive as CMI?

And why does Residence have the possibility to select via Worldedit?

**Comment by Zrips** [comment]
@blakedj012 
Why would you even need WE selection? You already have /cmi select command with most used function for area selection, like expanding, contracting or shifting. So why would we need option from WE? What does it add extra?
And in what way selection visualizer fails in comparison to other plugins? Its uses ingame particle effects and doesn't require any mods for you to have, which i suspect you are talking about when you mention better visualization.

**Comment by ghost** [comment]
Zrips why you have add it to residence ? Worldedit CUI Mod is ways better then any plugin.  And the Particles on Plugin WorldEditSUI better as you can edit it in Selection Visualizer.

**Comment by Zrips** [comment]
@blakedj012 
WorldEditSUI? Plugin which cant display bigger selection? SelectionVisualizer doesn't have any limits in how big selection is and you have complete ingame particle editor where you can define particles per player for each selection with bunch of customisations. So no idea why you compared with that plugin. 
Residence had We support for ages, but its not a main thing to have for plugin functionality and i always preferred none modded client support.
But in short, yes, we could add support for worldedit selections

**Comment by ghost** [comment]
Thanks that would make me and maybe others very happy. If it's not too much trouble. :)

**Comment by mrfloris** [comment]
WE CUI is limited to structure block distance which is 32 blocks, i think?
